### PR TITLE
ci: Pass down $MAKEJOBS to test_runner.py, other improvements

### PR DESCRIPTION
**Pull request by maflcko**
Some changes to the ci system:

* Pass down MAKEJOBS to the test_runner, instead of falling back to the default of 4. Passing it down avoids OOM on weak machines and allows better use of resources on beefy machines.
* Move CCACHE_DIR to ./ci/scratch/ subfolder: `ccache` is executed with root permissions inside the docker, so the cache files are created with root as owner. So it might be wise to not put them in the $HOME of the host
* Use the scratch dir as prefix for the test runner, as opposed to `/tmp/`, which is often a ramdisk and thus leads to OOM on the host when either a lot of tests are run in parallel or when a lot of tests fail and the datadirs are not cleaned.

### CODE REVIEW

File: ci/test/04_install.sh [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 export LC_ALL=C.UTF-8
 
+mkdir -p "${BASE_SCRATCH_DIR}"
```
Comment (promag) [inline ci/test/04_install.sh]: Is this really necessary? On my system:
```
$ ls /tmp/this
ls: /tmp/this: No such file or directory
$ CCACHE_DIR=/tmp/this/doesnt/exits ccache find /tmp/this
/tmp/this
/tmp/this/doesnt
/tmp/this/doesnt/exits
/tmp/this/doesnt/exits/9
/tmp/this/doesnt/exits/9/stats
/tmp/this/doesnt/exits/ccache.conf
```

Comment (promag) [reply ci/test/04_install.sh]: Oh right, when `CCACHE_DIR` is unrelated to `BASE_SCRATCH_DIR`, thanks.

**Review by fanquake — APPROVED** [review]
ACK fa27372e65359f12bea9ac0680d6bcb96680dda8

**Review by laanwj — APPROVED** [review]
ACK fa27372e65359f12bea9ac0680d6bcb96680dda8

**Pull request merged**